### PR TITLE
Delete button

### LINKED ISSUE #4: Add a way to delete color input fields
**Describe the solution you'd like**
A small close button in the top right corner of each input field that deletes it. This may require moving the circular color swatch slightly to accommodate the button

### PR DESCRIPTION
**Pull request by Hat52**
Added the delete button the the non default input fields

### CODE REVIEW

**Review by EmmaDawsonDev — CHANGES_REQUESTED** [review]
This Pr seems to contain some of the changes from your previous pull request about the reset button. Please update your branch so that the merged changes are reflected in this PR. Also, please revert the changes to the package-lock file. This should not be included in the pull request.

__
[sema-logo](https://semasoftware.com/gh) &nbsp;**Summary:** :hammer_and_wrench: This code needs a fix

File: src/components/ColorInput.jsx
Comment (EmmaDawsonDev) [inline src/components/ColorInput.jsx]: The icon itself only renders as an SVG and to be accessible to keyboard users it needs to wrapped in a button element. The icon itself would preferably be grey instead of red, so it doesn't stand out too much and with a white background so the input box border isn't visible through it. 

I also wonder about the logic. To me it would make more sense if when there are more than 2 inputs then all of them have a delete button so I can delete whichever color I wish, however when it gets down to only 2 inputs left then the buttons disappear. So instead of checking for inputId > 2 we check that the array length is > 2 instead.

__
[sema-logo](https://semasoftware.com/gh) &nbsp;**Summary:** :hammer_and_wrench: This code needs a fix

**Comment by Hat52** [comment]
I have implemented the changes. I tried to remove the file from the package-lock.json but I can't confirm it if it is removed or not can you confirm it and let me know.

File: src/App.css
Comment (EmmaDawsonDev) [inline src/App.css]: These changes are from the previous PR, I think you need to fetch upstream and merge it into your branch.

__
[sema-logo](https://semasoftware.com/gh) &nbsp;**Summary:** :hammer_and_wrench: This code needs a fix

File: src/App.css
Comment (EmmaDawsonDev) [inline src/App.css]:
```suggestion
  border-radius: var(--borderRadius);
```

File: src/components/ColorInput.jsx
Comment (EmmaDawsonDev) [inline src/components/ColorInput.jsx]: Please move this line of code to line 19 after the input element so it is inside the input container.

Also, the onClick handler needs moving from the icon to the button. 
```suggestion
      {(colorsLength > 2)?<button className="deleteButton" onClick={()=>handleDelete(inputId-1)}><AiOutlineCloseCircle color="grey"  /></button>:null}
```

__
[sema-logo](https://semasoftware.com/gh) &nbsp;**Summary:** :hammer_and_wrench: This code needs a fix

Comment (EmmaDawsonDev) [reply src/components/ColorInput.jsx]: Oh I nearly forgot - this button also needs an aria-label="delete input" as it doesn't have any discernable text that could be read aloud by a screen reader.

**Review by EmmaDawsonDev — CHANGES_REQUESTED** [review]
Great that you managed to remove the package lock file! 

Some of the changes from the previous pull request are still there which made it a bit difficult to review but I think I've managed to comment the most important things. In the future make sure to always make changes for different PRs on different branches so the two don't get muddled.

There's a few things to change before this is ready to merge, let me know if you need any help.

File: src/components/ColorInput.jsx
Comment (Hat52) [reply src/components/ColorInput.jsx]: okay I will start working on the changes.

Comment (Hat52) [reply src/components/ColorInput.jsx]: **Can you answer these question. It will make a lot of things clear to me**
- What is the best way to keep my fork updated with the main repo?
- Should I always make the new branch from the main branch (According to my knowledge this is the only way to all the branch separated)?
- Why do we need aria-label(what is its functionality)?

Comment (EmmaDawsonDev) [reply src/components/ColorInput.jsx]: At the top of your forked repo you should see a bar that looks like this:
image
If you click on fetch upstream it'll sync the two repositories. Then in your local clone you do git pull in the main branch. Then you change into your working branch and do git merge main and finally git push to sync your branch with your fork.

Yes, always make a new branch from main branch. You won't see some of the changes you made on another branch until the PR has been merged but it's the best way to avoid conflicts.

Because the button doesn't contain any text when a screen reader uses the site all they will hear is 'button' which doesn't give them any context about what the button does. If we add an aria label that will also be read out so they will instead hear 'button, delete input' and that will give them context about what happens when they click on the button.

Hope that helps!

Comment (Hat52) [reply src/components/ColorInput.jsx]: Yeah it made alot of sense. Thanks for the explanation 👍🙂

**Review by EmmaDawsonDev — APPROVED** [review]
this works nicely! Thank you for your contribution 😄 

__
[sema-logo](https://semasoftware.com/gh) &nbsp;**Summary:** :ok_hand: This code looks good

**Comment by Hat52** [comment]
Thanks

**Pull request merged**
